File: Java/Logic-2/closeFar/CloseFar.java
package closeFar;

public class CloseFar {
    //Retorne true se "b" ou "c" forem ao menos 1 unidade diferente de "a" e o outro for, pelo menos 2 ou mais unidades
    //diferente de ambos.
    //Ex.: ((4, 1, 3) → true; (1, 2, 3) → false).

    public boolean closeFar(int a, int b, int c) {
        int menor, maior;
        if (Math.abs(a-b) <= 1) {
            menor = b; maior = c;
        }else if (Math.abs(a-c) <= 1) {
            menor = c; maior = b;
        }else return false;
        return  (Math.abs(a-maior) >= 2 && Math.abs(menor-maior) >= 2);
    }
}

File: Python/Logic-2/close_far.py
"""
Dado 3 números inteiros ("a", "b" e "c"), retorne true se "b" ou "c" diferirem de
"a" por 1 ou menos e algum dos números diferir de 2 ou mais unidades em relação aos
outros.
"""


def close_far(a, b, c):
    if abs(a-b) <= 1:
        return abs(c-a) >= 2 and abs(b-c) >= 2
    if abs(a-c) <= 1:
        return abs(b-a) >= 2 and abs(b-c) >= 2
    return False


print(close_far(1, 2, 10))
